### ISSUE TITLE
Not authorized to edit Layers even as Admin

### ISSUE THREAD

**Issue by marcnyc2001** [craigk5n/webcalendar#369]
Hello, I am having an issue with LAYERS.
I am the ADMIN but when I click on ADMIN>Layers I get a message that says:
"!!!You are not authorized" and I am unable to add or even view the current layers.

Can somebody help me get access to this feature, please?
Thank you

**Comment by Tharrington86** [comment]
I actually discovered this same issue this morning. I tried other admin accounts and got the same message, effectively removing our ability to use layers at all.

**Comment by marcnyc2001** [comment]
Have you found a solution?

**Comment by craigk5n** [comment]
What version are you using.  I am unable to reproduce this in v1.9.9.

**Comment by marcnyc2001** [comment]
1.2.7

**Comment by Tharrington86** [comment]
I have not found a solution yet. I am using the latest version of the calendar. None of my admin accounts can access layers. They all receive the "You are not authorized!" message.

**Comment by marcnyc2001** [comment]
me neither. would love to find a solution to this!

**Comment by Tharrington86** [comment]
I tried to create a new Admin user and experienced the same issue. Here is a screen shot of what is seen in the console when the !!!You are not authorized message is displayed.

![image](https://github.com/craigk5n/webcalendar/assets/75510358/c8bdb85d-6394-440b-be83-28accda6b1cf)

**Comment by max951** [comment]
**Does your admin user have rights to layers?**

![hc_1322](https://github.com/craigk5n/webcalendar/assets/12292845/1687c845-cbfe-4723-a272-c77dcdd0be3f)

**Comment by marcnyc2001** [comment]
I thought mine does.
The issue is I get a "You are not authorized!!!" error so I can't even enable or disable them on my Admin page!

**Comment by max951** [comment]
I'm not sure exactly which page you mean that displays the authorization hint. What is the exact path in the address bar?

**Comment by marcnyc2001** [comment]
The link is: /calendar/layers.php
and the error is:
"!!!You are not authorized."
Written with 3 exclamation marks at the top of the sentence!

**Comment by max951** [comment]
If you have access to the database, can you please check if the the admin user has all access permissions.

You can do this in MySQL with the statement
`SELECT * FROM `webcal_access_function` WHERE cal_login = 'admin'`

or for example in phpMyAdmin, look in table webcal_access_function in the row for user admin.
There should be only 'Y's in the permissions column. Like in this Screenshot

![image](https://github.com/craigk5n/webcalendar/assets/12292845/e1aa3ac6-7fe8-42aa-b8cc-21dc92083968)

If there is a 'N' between the 'Y's, your admin user has not all permissions.

**Comment by marcnyc2001** [comment]
Yep, they are all set to "YYYYYYYYYYYYYYYYYYYYYYYYYYYY" on my Admin Account

**Comment by max951** [comment]
and the login name is 'admin'?

**Comment by marcnyc2001** [comment]
No the login name is my username (there is no username named "admin") but in the 'webcal_user' row the column 'cal_is_admin' is sedt to Y

**Comment by max951** [comment]
Sorry, but then I don't know what to do. I would try to debug starting from the layers.php file.

**Comment by marcnyc2001** [comment]
Thanks for your suggestion Max.
I tried that but I didn't see a different output in the browser.
In the error_log file there are a bunch of these lines:

`[09-Feb-2024 11:59:37 America/New_York] PHP Notice:  Use of undefined constant id - assumed 'id' in /home/eastside/public_html/calendar/includes/functions.php on line 4789
`

**Comment by max951** [comment]
The obove javascript debug line writes the output in the console, not in the browser itself.
You have to use the Browser DevTools ( [F12]-Key )

![hc_091](https://github.com/craigk5n/webcalendar/assets/12292845/0387cb7a-e6f2-4616-b49e-0530143f13e4)

I made this so complicated because the HTML output is not changed and other users are not disturbed by the debugging.


If you can change the HTML output without disturbing other users, then it is of course easier to simply use an echo command instead of the cryptic javascript debug line.

If you are using V1.2.7 the lines
`if ( $ALLOW_VIEW_OTHER != 'Y' )
  echo print_not_auth (7);`

should be in lines number 48 and 49.

I assume the line "echo print_not_auth" gives the error message "!!!You are not authorized" if the variable "$ALLOW_VIEW_OTHER" is not 'Y'.

To check this, you can simply insert the line
`echo '$ALLOW_VIEW_OTHER = ' . $ALLOW_VIEW_OTHER . "<br>\r\n";`
before the line with the if statement.

**Comment by marcnyc2001** [comment]
By adding that line I get:

$ALLOW_VIEW_OTHER = N
!!!You are not authorized.

**Comment by craigk5n** [comment]
The debugging in the thread found the answer -- ALLOW_VIEW_OTHER is set to N.  Layers let you overlay other users' calendars on yours, so they require that setting to be enabled.

To fix it, go to Admin > System Settings and set "Allow Viewing Other Users Calendars" to Yes.
